lib/straight-server/gateway.rb

```diff
@@ -32,15 +32,21 @@ def initialize_blockchain_adapters
       @blockchain_adapters = []
       StraightServer::Config.blockchain_adapters.each do |a|
 
-        adapter = Straight::Blockchain.const_get("#{a}Adapter")
+        adapter = Straight::Blockchain.const_get("#{a['name']}Adapter")
         next unless adapter
-        begin
-          main_url = StraightServer::Config.__send__("#{a.downcase}_url") rescue next
-          test_url = StraightServer::Config.__send__("#{a.downcase}_test_url") rescue nil
-          @blockchain_adapters << adapter.mainnet_adapter(main_url: main_url, test_url: test_url)
-        rescue ArgumentError
-          @blockchain_adapters << adapter.mainnet_adapter
-        end
+
+        @blockchain_adapters <<
+          case a['name']
+          when 'ChainCom'
+            adapter.mainnet_adapter(api_key_id: a['api_key_id'])
+          when 'Insight'
+            main_url = StraightServer::Config.__send__("#{a['name'].downcase}_url")
+            test_url = StraightServer::Config.__send__("#{a['name'].downcase}_test_url")
+            adapter.mainnet_adapter(main_url: main_url, test_url: test_url)
+          else
+            adapter.mainnet_adapter
+          end
```

Comment:
You right, it'll prettier. But I think, storing structures in config files - it isn't very good idea. If I want to know what settings I put in initializer, I'll have to look in the config file. I think, config files uses for keeping config data, not for application logic.

About `mainnet_adapter ` - I don't get what you want say me.